### ISSUE TITLE
Either export for Node/CommonJS or RequireJS, not both

### ISSUE THREAD

**Issue by pheyvaer** [jhlywa/chess.js#196]
At the moment the following happens:

```
/* export Chess object if using node or any other CommonJS compatible
 * environment */
if (typeof exports !== 'undefined') exports.Chess = Chess;
/* export Chess object for any RequireJS compatible environment */
if (typeof define !== 'undefined') define( function () { return Chess;  });
```

When using only Node this works fine, because the second statement is not executed. When using Webpack, however, both statements are executed (because it supports both if I'm correct), which means that the first statement is overwritten (at least it seems like that) and, hence, is therefore not longer compatible with code that is written for Node. And the code works without Webpack.
Note that I don't have this problem when using chess.js as a direct dependency, only when chess.js is an indirect dependency.

Therefore, I would suggest to turn these two separate ifs into in an if-else.

**Comment by ngocdaothanh** [comment]
Also, should we change:
`if (typeof exports !== 'undefined') exports.Chess = Chess;`
to:
`if (typeof module !== 'undefined') module.exports.Chess = Chess;`?

**Comment by maxwellhaydn** [comment]
@ngocdaothanh `module.exports` is unique to Node, so I think that would break with other CommonJS loaders.

**Comment by maxwellhaydn** [comment]
I need this enhancement. I’m using Webpack to bundle a React app, but my tests run in Node.

    const { Chess } = require('chess.js');

works in my tests, but not in my app.

The real issue is that CommonJS loaders return an object with the Chess constructor as a property, while AMD loaders return the Chess constructor directly. I was going to open a pull request to simply add the `else` as @pheyvaer suggested, but I think the real fix is to return the same thing for both CommonJS and AMD, e.g.

```
/* export Chess object if using node or any other CommonJS compatible
 * environment */
if (typeof exports !== 'undefined') {
    exports.default = Chess;
}
/* export Chess object for any RequireJS compatible environment */
else if (typeof define !== 'undefined') {
    define( function () {
        return Chess;
    });
}
```

That would be a breaking change to the interface, though, requiring CommonJS users to use a default import instead of a named import.

Of course, if there is any code out there where the tests use RequireJS but the code is bundled with Webpack, or maybe code that uses [RequireJS in Node](https://requirejs.org/docs/node.html), I think adding the `else` would break it.

**Comment by maxwellhaydn** [comment]
My current (ugly) workaround:

```
const { isNode } = require('browser-or-node');

let Chess;

if (isNode) {
    const chess = require('chess.js');
    Chess = chess.Chess;
}
else {
    Chess = require('chess.js');
}
```

**Comment by maxwellhaydn** [comment]
It turns out there's an easy way to fix this with Webpack. Simply disable the AMD parser for chess.js in your Webpack config:

    module: {
        rules: [
        ...
            {
                test: require.resolve('chess.js'),
                parser: {
                    amd: false
                }
            }
        ]
    }

Now you can run

    const { Chess } = require('chess.js');

in both your Node code and your Webpack-bundled code.

**Comment by jhlywa** [comment]
Fixed in 0.13.0
